src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+interface TodoModalProps {
+  onClose: () => void;
+  user: User | null;
+}
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+export const TodoModal: React.FC<TodoModalProps> = ({ onClose, user }) => {
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const [loading, setLoading] = useState(true);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (currentTodo) {
+        setLoading(false);
+      }
+    }, 300);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    return () => clearTimeout(timer);
+  }, [currentTodo]);
```

Comment:
You are using `setTimeout` to simulate loading state. While this can work for development or testing, it's not a good practice for production code. It would be better to set the loading state based on actual data fetching process. For instance, set `loading` to true before you start fetching data and set it to false when the data is received.